src/topology/locally_constant/algebra.lean

```diff
@@ -110,4 +110,29 @@ instance [ring Y] : ring (locally_constant X Y) :=
 instance [comm_ring Y] : comm_ring (locally_constant X Y) :=
 { .. locally_constant.comm_semiring, .. locally_constant.ring }
 
+variables (A : Type*) [semiring A]
+
+instance : has_scalar A (locally_constant X A) :=
+{ smul := λ a f,
+  { to_fun := λ x, a*f(x),
+    is_locally_constant := is_locally_constant.comp (is_locally_constant f)
+      (has_mul.mul a) } }
+
+instance : mul_action A (locally_constant X A) :=
+{ smul := (•),
+  one_smul := one_mul,
+  mul_smul := λ a b f, ext ( λ x, show a * b * f x = a * (b * f x), from mul_assoc _ _ _ ) }
+
+instance : distrib_mul_action A (locally_constant X A) :=
+{
+  smul_add := λ r f g, ext ( λ x, show  r * (f x + g x) = r * f x + r * g x, from mul_add _ _ _ ),
```

Comment:
I don't know what the preferred style is, but you could delete `show ..., from` the three times you use it and things will still compile, because Lean can figure out what you want to prove from the proof you give.

Reply:
Indeed it works, thank you!